include/cppkafka/clonable_ptr.h

```diff
@@ -91,13 +91,41 @@ class ClonablePtr {
         return handle_.get();
     }
     
+    /**
+     * \brief Clones the internal pointer using the specified cloner function.
+     */
+    T* clone() const {
```

Comment:
Do you think it makes sense to expose this member function? Shouldn't it be private given you get the same, but safer, behavior if you use the copy constructor?

Reply:
I will make it private. Initially i was using it in the BasicMessageBuilder constructor when you from one type to another. The normal copy construction doesn't work there and you have to work directly by cloning the underlying handle (or calling `rd_kafka_headers_copy`).  Reason I removed it is because there is no way to know if calling `clone()` will be a no-op or a real copy.